src/libstore/profiles.hh

```diff
@@ -37,6 +37,8 @@ void deleteGeneration(const Path & profile, unsigned int gen);
 
 void deleteGenerations(const Path & profile, const std::set<unsigned int> & gensToDelete, bool dryRun);
 
+void deleteGenerationsGreaterThan(const Path & profile, const string & max, bool dryRun);
```

Comment:
`max` should be an integer. (I.e. the conversion to int should be done in nix-env.cc where the argument parsing happens.)